addons/mail_plugin/controllers/mail_plugin.py

```diff
@@ -152,10 +151,25 @@ def res_partner_get(self, email=None, name=None, partner_id=None, **kwargs):
         if not normalized_email:
             return {'error': _('Bad Email.')}
 
-        # Search for the partner based on the email.
-        # If multiple are found, take the first one.
-        partner = request.env['res.partner'].search(['|', ('email', 'in', [normalized_email, email]),
-                                                     ('email_normalized', '=', normalized_email)], limit=1)
+        notification_email = request.env['ir.mail_server']._get_default_from_address()
+        if normalized_email == tools.email_normalize(notification_email):
+            return {
+                'partner': {
+                    'name': _('Notification'),
+                    'email': normalized_email,
+                    'enrichment_info': {
+                        'type': 'odoo_custom_error', 'info': _('This is your notification address. Search the Contact manually to link this email to a record.'),
+                    },
+                },
+            }
+        else:
+            # Search for the partner based on the email.
+            # If multiple are found, take the first one.
+            partner = request.env['res.partner'].search(
+                ['|', ('email', 'in', [normalized_email, email]),
+                 ('email_normalized', '=', normalized_email)],
+                limit=1,
+            )
```

Comment:
Would avoid putting this inside a "else", as it's a pre-condition, a bit like the "if not normalized_email" above.

In addition, it will allow to not modify that part, making it a smaller diff for an old stable.

Reply:
ah indeed, good catch